modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2903,9 +2903,16 @@ private Predicate _getRelationshipObjectFieldPredicate(
 			return null;
 		}
 
+		Column<?, Long> primaryKeyColumn = (Column<?, Long>)table.getColumn(
+			objectDefinition.getPKObjectFieldDBColumnName());
+
+		if (primaryKeyColumn == null) {
```

Comment:
Hey @linolaoj , I think it would be better to reuse this logic, this way we concentrate this in just one place that will also help our code maintenance.
We have something similar [here](https://github.com/liferay/liferay-portal/blob/7c720a2bc9c2a31d7f3ecd5144844851331557c4/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2853-L2868). In your case it would be something like:

```
	private Column<?, Long> _getPrimaryKeyColumn(
		String pkObjectFieldDBColumnName, Table<?> table) {

		Column<?, Long> primaryKeyColumn = (Column<?, Long>)table.getColumn(
			pkObjectFieldDBColumnName);

		if (primaryKeyColumn == null) {
			primaryKeyColumn = ObjectEntryTable.INSTANCE.objectEntryId;
		}

		return primaryKeyColumn;
	}
```

Reply:
Hi @guilhermedcamacho, I made this change here https://github.com/liferay-objects/liferay-portal/pull/3081/commits/e31d1019ed27d61b8f4f30c471eb0c12f7ad8320.
Thanks!